src/opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.cpp

```diff
@@ -494,6 +501,12 @@ void RestartConfig::handleScheduleSection(const SCHEDULESection& schedule) {
         if (0 == timestep)
             return m_write_initial_RST_file;
 
+        if (save_keywords[timestep]) {
```

Comment:
I didn't think this through before, but I'm getting test failures running #454 after this change.  Specifically, the `Ewoms::EclOutputBlackOilModule<>::allocBuffers()` function [calls](https://github.com/OPM/ewoms/blob/master/ebos/ecloutputblackoilmodule.hh#L219-L221) `Opm::RestartConfig::getWriteRestartFile()` with a `timestep` that is equal to [`simulator_.episodeIndex() + 1`](https://github.com/OPM/ewoms/blob/master/ebos/eclwriter.hh#L241-L245).  In a simulation run for which
```
m_timemap.numTimesteps() == 39
```
that in turn means trying to access
```
save_keywords[39]
```
which is undefined behaviour.

Since there already is a special case treatment of `timestep == 0`, does that mean that we should be using
```C++
if (save_keywords[timestep - 1]) {
```
here?

Reply:
> Try [OPM/opm-common#499](https://github.com/OPM/opm-common/pull/499)

That solves my immediate problem in the context of #454.  However, now I get test failures like
```
The following tests FAILED:
	  6 - EclipseStateTests (Failed)
	  7 - BoxTest (Failed)
	 14 - ParseKEYWORD (Failed)
	 81 - RestartConfigTests (Failed)
	 90 - SummaryConfigTests (Failed)
	108 - EclipseIO (Failed)
Errors while running CTest
```
I get these both on the master branch and when testing #454.  These all happen in `Opm::RestartConfig::handleScheduleSection()`&mdash;specifically in the section that [initializes](https://github.com/OPM/opm-common/blob/master/src/opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.cpp#L400-L404) the `save_keywords` vector&mdash;and are all of the form
```
attempt to subscript container 
    with out-of-bounds index 4, but container only holds 4 elements.
```

Reply:
I am not able to reproduce the test failures when testing #454. I pulled the current state of the #454 branch is there any unpushed changes?

Reply:
> I pulled the current state of the #454 branch is there any unpushed changes?

There are not and this issue presents on master too.  You need libstdc++'s ["Debug Mode"](https://gcc.gnu.org/onlinedocs/libstdc++/manual/debug_mode.html) using flags like
```
-D_GLIBCXX_DEBUG=1  -D_GLIBCXX_DEBUG_PEDANTIC=1
```
to see it though.  If you don't want to go through the complete dance of adding those flags&mdash;which change the ABI so you need to apply them throughout all modules, including Dune&mdash;you can do something like
```diff
diff --git a/src/opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.cpp b/src/opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.cpp
index 5213985c..e13948fb 100644
--- a/src/opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.cpp
+++ b/src/opm/parser/eclipse/EclipseState/IOConfig/RestartConfig.cpp
@@ -398,9 +398,9 @@ void RestartConfig::handleScheduleSection(const SCHEDULESection& schedule) {
         }
 
         if (name == "SAVE") {
-            this->save_keywords[current_step] = true;
+            this->save_keywords.at(current_step) = true;
         } else {
-            this->save_keywords[current_step]= false;
+            this->save_keywords.at(current_step) = false;
         }
 
         if( !( name == "RPTRST" || name == "RPTSCHED" ) ) continue;
```
in which case you'll see exceptions of the form
```
unknown location(0): fatal error: in "TestIOConfigCreationWithSolutionRPTSOL": std::out_of_range: vector<bool>::_M_range_check: __n (which is 4) >= this->size() (which is 4)
/home/bska/work/opm/src/opm/opm-common/tests/parser/EclipseStateTests.cpp(558): last checkpoint: "TestIOConfigCreationWithSolutionRPTSOL" entry.
```

Reply:
Thanks for the explanation. I have updated OPM/opm-common#499.